src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/register/PullRequestStoreImpl.java

```diff
@@ -0,0 +1,91 @@
+package com.atlassian.bitbucket.jenkins.internal.trigger.register;
+
+import com.atlassian.bitbucket.jenkins.internal.model.BitbucketPullRequest;
+import com.atlassian.bitbucket.jenkins.internal.scm.BitbucketSCMRepository;
+
+import javax.inject.Singleton;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ConcurrentMap;
+
+@Singleton
+public class PullRequestStoreImpl implements PullRequestStore {
+
+    private final ConcurrentMap<PullRequestStoreImpl.CacheKey, ConcurrentLinkedQueue<BitbucketPullRequest>> pullRequests;
```

Comment:
They are not persisted. They are just like normal maps, but threadsafe. 
You raise a valid question, and my memory was that we will write a "refreshPrs" or some such thing. It will run on startup, and then regularly after that.